agent/systemd/notify.go

```diff
@@ -0,0 +1,33 @@
+package systemd
+
+import (
+	"errors"
+	"net"
+	"os"
+)
+
+var NotifyNoSocket = errors.New("No socket")
+
+// Notifier provides a method to send a message to systemd.
+type Notifier struct{}
+
+// Notify sends a message to the init daemon. It is common to ignore the error.
+func (n *Notifier) Notify(state string) error {
+	addr := &net.UnixAddr{
+		Name: os.Getenv("NOTIFY_SOCKET"),
```

Comment:
I don't think so since the name of the env var is fixed when the type is `notify`. I've previously missed that `READY=1` is the value we are supposed to send when the service is up so it may make sense to define some constants.

https://www.freedesktop.org/software/systemd/man/sd_notify.html